packages/admin/src/pages/build-application/[applicationId]/[sectionId]/question-type/index.getServerSideProps.tsx

```diff
@@ -0,0 +1,142 @@
+import { GetServerSideProps, GetServerSidePropsContext } from 'next';
+import ResponseType from '../../../../../enums/ResponseType';
+import { getSessionIdFromCookies } from '../../../../../utils/session';
+import { questionErrorPageRedirect } from '../newQuestionServiceError';
+import {
+  addFieldsToSession,
+  getSummaryFromSession,
+  getValueFromSession,
+} from '../../../../../services/SessionService';
+import { QuestionSummary } from '../../../../../types/QuestionSummary';
+import { postQuestion } from '../../../../../services/QuestionService';
+import { getApplicationFormSummary } from '../../../../../services/ApplicationService';
+import QuestionPageGetServerSideProps from '../../../../../utils/QuestionPageGetServerSideProps';
+import ResponseTypeEnum from '../../../../../enums/ResponseType';
+
+type RequestBody = {
+  responseType: ResponseType;
+  _csrf?: string;
+};
+
+const redirectQuestionType = [
+  ResponseType.Dropdown,
+  ResponseType.MultipleSelection,
+  ResponseType.LongAnswer,
+];
+
+function getRedirect(
+  responseType: ResponseType,
+  applicationId: string,
+  sectionId: string
+) {
+  const REDIRECT_MAP = {
+    [ResponseType.Dropdown]: `/build-application/${applicationId}/${sectionId}/question-options`,
+    [ResponseType.MultipleSelection]: `/build-application/${applicationId}/${sectionId}/question-options`,
+    [ResponseType.LongAnswer]: `/build-application/${applicationId}/${sectionId}/question-type/add-word-count`,
+  };
+  return REDIRECT_MAP[responseType as keyof typeof REDIRECT_MAP];
+}
+
+const SHORT_QUESTION_WORD_LIMIT = 300;
+
+function getRadio(responseType: ResponseType) {
+  if (responseType === ResponseType.Dropdown) return 'Multiple choice';
+  if (responseType === ResponseType.MultipleSelection) return 'Multiple select';
+  return '';
+}
+
+export const getServerSideProps = async (
+  context: GetServerSidePropsContext
+) => {
+  const { params, req } = context;
+  const { applicationId, sectionId } = params as Record<string, string>;
+
+  const sessionId = req.cookies.session_id;
+  const sessionCookie = getSessionIdFromCookies(req);
+
+  if (!sessionId) {
+    // We expect the session ID to be present in order to get to this page
+    return questionErrorPageRedirect(applicationId);
+  }
```

Comment:
Wont middleware guarentee sessionId exists on this page?

Reply:
actually hol' up whats the difference between `sessionId` & `sessionCookie`?

Reply:
> hol' up whats the difference between sessionId & sessionCookie?
literally nothing as it turns out. Will rm